Helper/AnalyticsHelper.php

```diff
@@ -0,0 +1,380 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Helper;
+
+use Algolia\AlgoliaSearch\Helper\Entity\CategoryHelper;
+use Algolia\AlgoliaSearch\Helper\Entity\PageHelper;
+use Algolia\AlgoliaSearch\Helper\Entity\ProductHelper;
+use AlgoliaSearch\Analytics;
+
+class AnalyticsHelper extends Analytics
+{
+    const ANALYTICS_SEARCH_PATH = '/2/searches';
+    const ANALYTICS_HITS_PATH = '/2/hits';
+    const ANALYTICS_FILTER_PATH = '/2/filters';
+    const ANALYTICS_CLICKS_PATH = '/2/clicks';
+
+    const INTERNAL_API_PROXY_URL = 'https://magento-proxy.algolia.com/';
+
+    /** @var \Algolia\AlgoliaSearch\Helper\AlgoliaHelper */
+    private $algoliaHelper;
+
+    /** @var \Algolia\AlgoliaSearch\Helper\ConfigHelper */
+    private $configHelper;
+
+    /** @var Data */
+    private $dataHelper;
+
+    /** @var Product */
+    private $productHelper;
+
+    /** @var CategoryHelper */
+    private $categoryHelper;
+
+    /** @var PageHelper */
+    private $pageHelper;
+
+    private $logger;
+
+    /** Cache variables to prevent excessive calls */
+    private $searches;
+    private $users;
+    private $rateOfNoResults;
+
+    private $clickPositions;
+    private $clickThroughs;
+    private $conversions;
+
+    private $clientData;
+
+    private $errors = [];
+
+    /**
+     * AnalyticsHelper constructor.
+     * @param AlgoliaHelper $algoliaHelper
+     * @param ConfigHelper $configHelper
+     * @param Data $dataHelper
+     * @param ProductHelper $productHelper
+     * @param CategoryHelper $categoryHelper
+     * @param PageHelper $pageHelper
+     * @param Logger $logger
+     */
+    public function __construct(
+        AlgoliaHelper $algoliaHelper,
+        ConfigHelper $configHelper,
+        Data $dataHelper,
```

Comment:
`DataHelper`, `ProductHelper`, `CategoryHelper`, `PageHelper` are injected only to provision index names. How about extract it to a standalone `AnalyticsIndexNameProvider` (just example name) class and lower number of dependencies here? :)